tensornetwork/tensornetwork.py

```diff
@@ -966,6 +966,44 @@ def contract_parallel(self, edge: Edge) -> Node:
       raise ValueError("Attempted to contract dangling edge: '{}'".format(edge))
     return self.contract_between(edge.node1, edge.node2)
 
+  def squeeze(self, edge: Edge) -> Node:
+    """Squeezes a dangling edge with unit size.
```

Comment:
Also say that this removes the edge from the node.

Reply:
Why restrict this to dangling edges?

Reply:
This basically just removes unit sized dangling edges. Non-dangling unit sized edges are either outer products or no-op contractions.

Reply:
Yes, I know. I think it makes just as much sense to squeeze them away as it does to squeeze dangling edges.